Tribler/Core/Modules/tracker_manager.py

```diff
@@ -95,6 +95,11 @@ def update_tracker_info(self, tracker_url, is_successful):
         :param tracker_url: The given tracker_url.
         :param is_successful: If the check was successful.
         """
+        # It is possible that a Deferred call could not be cancelled and invokes this function while
+        # we are shutting down, in that case we ignore it.
+        if not self._tracker_dict:
```

Comment:
It won't because the delayed call has been canceled with cancel_all_pending_tasks()